onedal/primitives/tree_visitor.cpp

```diff
@@ -0,0 +1,344 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "onedal/common/pybind11_helpers.hpp"
+#include "onedal/common.hpp"
+#include "oneapi/dal/algo/decision_forest.hpp"
+#include "numpy/arrayobject.h"
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <algorithm>
+#include <limits>
+#include <vector>
+
+#include <iostream>
+#include <utility>
+
+#define TERMINAL_NODE -1
+#define NO_FEATURE    -2
```

Comment:
Don't we have some rules on macros naming?
Like that macros should start with "ONEDAL_PY_"